### ISSUE TITLE
Specify different maximum channel number for user, moderator and administrator

### ISSUE THREAD

**Issue by EricGuic** [Chocobozzz/PeerTube#5431]
### Describe the problem to be solved

I would like the option to specify a different maximum number of channel a user can create, regarding his type.

### Describe the solution you would like

In the Administration/Configuration/Basic tab, then in the Vidéo channel section, add two more fields (for moderator and administrator) so we can specify a different number for each type of user.

This would be the "by default" value.
And even better, being able to overwrite this value at the user level.

**Comment by EricGuic** [comment]
Hi Chocobozzz,
Just trying to understand the Label [Component: PeerTube Plugin] : you consider that this features should be implemented through a plugin, and therefore will not be on your to-do-list, is that right ? (no critic at all, it's just to know if I have to manage it by myself ;-)
Best regards, Eric.

**Comment by Chocobozzz** [comment]
Yes, but we need to add hooks to our plugin API so you can implement this feature.

I'm not against to add this feature in PeerTube core, but we won't have time to develop this feature this year.
I also think it can be part of a bigger "profile" feature, where you would assign settings (max channels, video import enabled, live enabled etc) to a group of users by assigning them a custom profile.

**Comment by EricGuic** [comment]
Hi, is there another issue related to this potential "profil" feature, to see the progression and maybe add some ideas like this one ?

**Comment by EricGuic** [comment]
Hi, is there any update on a bigger "profile" feature, which could resolve this issue ? Thanks.

**Comment by Chocobozzz** [comment]
> Hi, is there any update on a bigger "profile" feature, which could resolve this issue ? Thanks.

No sorry, but don't hesitate to contact Framasoft if you want to help us to fund this feature. I also created the profile feature request in https://github.com/Chocobozzz/PeerTube/issues/6668
